mender-auth/main.cpp

```diff
@@ -31,14 +35,54 @@ namespace http = mender::http;
 namespace ipc = mender::auth::ipc;
 namespace mlog = mender::common::log;
 namespace setup = mender::common::setup;
+namespace crypto = mender::common::crypto;
+namespace expected = mender::common::expected;
+namespace io = mender::common::io;
+namespace path = mender::common::path;
+
+static expected::ExpectedString GetPassphraseFromFile(const string &filepath) {
+	string passphrase = "";
+	if (filepath == "") {
+		return passphrase;
+	}
+
+	auto ex_ifs = io::OpenIfstream(filepath == "-" ? "/dev/stdin" : filepath);
+	if (!ex_ifs) {
+		return expected::unexpected(ex_ifs.error());
+	}
+	auto &ifs = ex_ifs.value();
+
+	errno = 0;
+	getline(ifs, passphrase);
+	if (ifs.bad()) {
+		int io_errno = errno;
+		error::Error err {
+			generic_category().default_error_condition(io_errno),
+			"Failed to read passphrase from '" + filepath + "'"};
+		return expected::unexpected(err);
+	}
+
+	return passphrase;
+}
 
 int main(int argc, char *argv[]) {
 	setup::GlobalSetup();
 
+	string passphrase_file = "";
 	conf::MenderConfig config;
 	if (argc > 1) {
 		vector<string> args(argv + 1, argv + argc);
-		auto success = config.ProcessCmdlineArgs(args.begin(), args.end());
+
+		vector<struct conf::ExtraOption> extra = {
+			{"--passphrase-file",
+			 [&passphrase_file](const string &option, const string &value) {
+				 if (option == "--passphrase-file") {
+					 passphrase_file = value;
+				 }
+			 },
+			 true}};
+
+		auto success = config.ProcessCmdlineArgs(args.begin(), args.end(), extra);
```

Comment:
With this the option is being added to the top level, correct? It should only be added to the `daemon` command, so IOW:

```bash
mender-client --passphrase-file myfile daemon # WRONG
mender-client daemon --passphrase-file myfile # Correct
```

If it's done like this, then I believe you don't need any changes to `conf.cpp` at all, because you can just do the same thing which is done [in mender-update's `cli.cpp`](https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/mender-update/cli/cli.cpp). Notice how `ProcessCmdlineArgs` is called without any extra arguments, and then extra arguments are parsed afterwards in `ParseUpdateArguments` because the former stops parsing after the command name.

**Update**: Ok, so after writing this, I realized that we *don't have* a `daemon` command, because [MEN-6639](https://northerntech.atlassian.net/browse/MEN-6639) hasn't been done yet! Sorry, this little detail slipped my mind. But that ticket should be very easy, perhaps you can just combine the two so that we do it right the first time?